python/ray/data/_internal/planner/exchange/aggregate_task_spec.py

```diff
@@ -65,10 +65,18 @@ def map(
         ]
         from ray.data.block import BlockMetadataWithSchema
 
-        meta_with_schema = BlockMetadataWithSchema.from_block(
-            block, stats=stats.build()
-        )
-        return parts + [meta_with_schema]
+        assert len(parts) > 0
```

Comment:
If the block is empty, this will be empty as well.

Let's for now handle empty blocks (please also add todo to clean this up once we fix empty blocks issue)